workspace/utilities/lib/struct-modal.xsl

```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+<!-- 
+	Creates a modal, either for a page or a simple element.
+
+	Content
+		modal: The actual content of the modal
+		close: The element that will trigger a closing action
+
+	Options
+		id: Unique identifier used to easily target a specific modal
+		is-visible: Determines if the modal starts visible or not
+		is-page: Changes the behavior of the modal to make it work as a page
+
+	Dependencies
+		modules/change-state.js
+		modules/auto-change-state-click.js
+-->
+
+
+<!-- COMPONENT: struct-modal -->
+	<xsl:template name="struct-modal">
+		<xsl:param name="content-modal" />
+		<xsl:param name="content-close" />
+		<xsl:param name="id" />
+		<xsl:param name="is-visible" select="false()" />
+		<xsl:param name="is-page" select="false()" />
+		<xsl:param name="ext-attr" />
+		<xsl:param name="ext-attr-background" />
+		<xsl:param name="ext-attr-ctn" />
+		<xsl:param name="ext-attr-close-ctn" />
+
+		<!-- For background and close button -->
+		<xsl:variable name="util-close-attr">
+			<xsl:choose>
+				<xsl:when test="$is-page">
+					<add href="{/data/params/current-path}/" />
+					<add data-action="toggle" />
+				</xsl:when>
+				<xsl:otherwise>
+					<!-- STATE (click): visible ~> OFF -->
+					<add class="js-change-state-click" />
+					<add data-change-state-click-target=".js-struct-modal[data-id='{$id}']" />
+					<add data-change-state-click="visible" />
+					<add data-change-state-action="off" />
+					<add data-change-state-click-target-common-ancestor="#site" />
+				</xsl:otherwise>
+			</xsl:choose>
+		</xsl:variable>
+
+		<!-- Element type of background -->
+		<xsl:variable name="element-background">
+			<xsl:choose>
+				<xsl:when test="$is-page"><xsl:text>a</xsl:text></xsl:when>
+				<xsl:otherwise><xsl:text>button</xsl:text></xsl:otherwise>
```

Comment:
`<xsl:text>` is only usefull when multiline. In this case, I would drop it, or indent it.

Reply:
Done!